power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -221,35 +216,26 @@ std::vector<Idx2D> get_component_sequence(MainModelState<ComponentContainer> con
 } // namespace detail
 
 template <class... ComponentTypes, class ComponentContainer>
-main_core::utils::SequenceIdx<ComponentTypes...>
+utils::SequenceIdx<ComponentTypes...>
 get_all_sequence_idx_map(MainModelState<ComponentContainer> const& state, ConstDataset const& update_data,
-                         Idx scenario_idx,
-                         main_core::utils::ComponentFlags<ComponentTypes...> const& components_to_store) {
+                         Idx scenario_idx, utils::ComponentFlags<ComponentTypes...> const& components_to_store,
+                         independence::UpdateIndependence<ComponentTypes...> const& independence, bool cached) {
     return utils::run_functor_with_all_types_return_array<ComponentTypes...>(
-        [&update_data, &components_to_store, &state, scenario_idx]<typename CompType>() {
-            if (!std::get<main_core::utils::index_of_component<CompType, ComponentTypes...>>(components_to_store)) {
+        [&state, &update_data, scenario_idx, &components_to_store, &independence, cached]<typename CompType>() {
+            auto const component_properties =
+                std::get<utils::index_of_component<CompType, ComponentTypes...>>(independence);
+            // The sequence for the independent components is cached (true). For the remaining components, the sequence
+            // cannot be cached (false), so the independence flags are inverted to not return an empty sequence when
+            // this is the case.
+            bool const component_independence =
+                !cached ? component_properties.is_independent() : !component_properties.is_independent();
```

Comment:
```suggestion
            bool const component_independence = cached ^ component_properties.is_independent();
```

Reply:
bitwise XOR is quite uncommon in modern C++ (except in microcontroller/compiler development), to the point that most new developers don't even know it exists. In particular, it is dangerous because `true` can be represented by any non-zero value. Please use `!=` instead

```suggestion
            bool const component_independence = cached != component_properties.is_independent();
```

Reply:
> In particular, it is dangerous because

Arguably anything is dangerous given certain context. In this particular case, the operands are two booleans. 

> bitwise XOR is quite uncommon in modern C++ (except in microcontroller/compiler development), to the point that most new developers don't even know it exists.

One could make an analogical argument that "arithmetic is quite uncommon in modern education, to the point that most new students don't even know how to do it". It is far fetched, funny but probably worth thinking.

Reply:
Since both suggestions are equally valid in this specific case, hence a matter of style, I went with `bool const component_independence = cached != component_properties.is_independent();` as it is more straightforward for me.